python/pyarrow/dataset.py

```diff
@@ -433,6 +433,10 @@ def _filesystem_dataset(source, schema=None, filesystem=None,
     FileSystemDataset
     """
     format = _ensure_format(format or 'parquet')
+    if isinstance(format, CsvFileFormat):
```

Comment:
hmm, it feels like this needs to be placed "lower" (unless there aren't any other ways of building a filesystem dataset?)

Reply:
I can move this down to the place where the `CsvFragmentScanOptions` is created, and just set the wrapper function in the init function. That also takes care of there being a publicly accessible function which we don't want.

Reply:
I think that makes sense.

Reply:
Done in [9a5bf20](https://github.com/apache/arrow/pull/13709/commits/9a5bf209a859fd05d5d219421a7ec6cbc205f28f)